fuel-vm/src/constraints.rs

```diff
@@ -94,15 +94,19 @@ impl CheckedMemRange {
         if constraint.end > VM_MAX_RAM {
             return Err(Bug::new(BugId::ID009, BugVariant::InvalidMemoryConstraint).into());
         }
-        Self::new_inner(address, size, constraint)
+        Self::new_inner(address as usize, size, constraint)
     }
 
     /// Create a new memory range, checks that the range is not empty
```

Comment:
This comment is no longer correct.

Does this change break the specs for other opcodes unrelated to SMO, such as MCP? I didn't see anything specifically disallowing this from the specs on cursory glance.

cc @Dentosal

Reply:
This shouldn't break anything there. Zero-length MCP is allowed by the spec, and should work correctly in the VM ([test case](https://github.com/FuelLabs/fuel-vm/blob/62d35e57f69d31d8bfa014e4f2f200e8e9e4fbd0/fuel-vm/src/tests/memory.rs#L272)).